api-tests/src/test/java/org/openmrs/module/reporting/report/definition/service/ReportDefinitionServiceImplTest.java

```diff
@@ -39,11 +40,32 @@
 	@Test
 	public void purgeDefinition_shouldPurgeReportDesigns() throws Exception {
 		executeDataSet("org/openmrs/module/reporting/include/ReportDefinitionServiceImplTest.xml");
-    	
+
     	ReportDefinition definition = reportDefinitionService.getDefinition(80);
     	reportDefinitionService.purgeDefinition(definition);
 	    
-    	ReportDesign reportDesign = reportService.getReportDesign(2);
-    	assertThat(reportDesign, nullValue());
+		assertThat(reportService.getReportDesign(3), nullValue());
+		assertThat(reportService.getReportDesign(4), nullValue());
+	}
+
+	/**
+	 * @see ReportDefinitionServiceImpl#purgeDefinition(String)
+	 * @verifies purge report definition by uuid and associated report designs and requests
+	 */
+	@Test
+	public void purgeDefinition_shouldPurgeDesignsAndRequestsAndDefinition() throws Exception {
+		executeDataSet("org/openmrs/module/reporting/include/ReportDefinitionServiceImplTest.xml");
+		assertNotNull(reportService.getReportDesign(3));

```

Comment:
Since you are purging a definition by uuid, your assertions would be clearer if they also fetch by uuid

Reply:
Good point, I will try to refactor 💯